common/predictive-text/unit_tests/headless/default-word-breaker.js

```diff
@@ -19,4 +20,63 @@ describe('The default word breaker', function () {
       "let's", 'eat', 'phở', '!', '🥣'
     ]);
   });
+
+  // The following tests are performed with model integration as an internal
+  // test for the wordbreaking API.
+  it('recognizes a word at end of lefthand context', function () {
+    var model = new TrieModel(jsonFixture('tries/english-1000'));
+
+    // Standard case - wordbreaking at the end of a word.
+    var context = { 
+      left: 'The quick brown fox jumped', startOfBuffer: true,
+      right: ' over the lazy dog.', endOfBuffer: true
+    };
+
+    var broken = model.wordbreak(context);
+
+    assert.strictEqual(broken, 'jumped');
+  });
+
+  it('returns text for a word in-progress', function() {
+    var model = new TrieModel(jsonFixture('tries/english-1000'));
+
+    // Standard case - midword (xylophone) call
+    var context = { 
+      left: 'xyl', startOfBuffer: true,
+      right: '', endOfBuffer: true
+    };
+
+    var broken = model.wordbreak(context);
+
+    assert.strictEqual(broken, 'xyl');
+  });
+    
+  it('returns empty string when called without word text', function() {
+    var model = new TrieModel(jsonFixture('tries/english-1000'));
+
+    // Wordbreaking on a empty space => no word.
+    context = { 
+      left: 'The quick brown fox jumped ', startOfBuffer: true,
+      right: 'over the lazy dog.', endOfBuffer: true
+    };
+
+    broken = model.wordbreak(context);
+
+    assert.strictEqual(broken, '');
+  });
```

Comment:
One more corner-case test may be nice: what should be returned when the buffer is completely empty?